Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -45,10 +45,15 @@ public class NetworkManager : MonoBehaviour
 
         /// <summary>
         /// Automatically invoke StartServer()
-        /// <para>If the application is a Server Build or run with the -batchMode command line arguement, StartServer is automatically invoked.</para>
+        /// <para>If the application is a Server Build, StartServer is automatically invoked.</para>
+        /// <para>Server build is true when "Server build" is checked in build menu, or BuildOptions.EnableHeadlessMode flag is in BuildOptions</para>	
         /// </summary>
-        [Tooltip("Should the server auto-start when the game is started in a headless build?")]
-        public bool startOnHeadless = true;
+        [Tooltip("Should the server auto-start when 'Server Build' is checked in build settings")]
+        [FormerlySerializedAs("startOnHeadless")]
+        public bool startOnServerBuild = true;
```

Comment:
is this the most obvious name?
what about startIfServerbuild or autoStartServerBuild (or keeping startOnheadless)?

Reply:
autoStartServerBuild sounds nice too